packages/app/components/Queue/TA/TAQueue.tsx

```diff
@@ -142,14 +142,16 @@ export default function TAQueue({ qid, courseId }: TAQueueProps): ReactElement {
                     Help Next
                   </HelpNextButton>
                 </Tooltip>
+
                 <div style={{ marginBottom: "12px" }}>
                   <TACheckinButton
                     courseId={courseId}
                     room={queue?.room}
                     disabled={
                       staffCheckedIntoAnotherQueue ||
                       isHelping ||
-                      (queue.isProfessorQueue && role !== Role.PROFESSOR)
+                      (queue.isProfessorQueue && role !== Role.PROFESSOR) ||
+                      queue.isDisabled
```

Comment:
Could you also do this disabling for the `Join` button that students get (its in `StudentQueue.tsx`) 🙏